src/lithopscloud/modules/cloud_functions/cloud_function.py

```diff
@@ -0,0 +1,147 @@
+        res = conn.getresponse()
+        metadata = res.read().decode("utf-8")
+        json_struct = json.loads(metadata)  # turn string to a dictionary
+
+        return json_struct
+
+    @spinner
+    def get_cloud_function_namespaces(self, selected_region):
+        """returns the names of the namespaces within a given region."""
+        msg = f"Obtaining Cloud Function namespaces in {selected_region}" if selected_region \
+            else "Obtaining all existing Cloud Function namespaces"
+        print(msg + '...\n')
+
+        namespaces = {}
+
+        for region in [selected_region] if selected_region else CF_REGIONS:
+            collecting_namespaces = True
+            max_limit = 200
+            offset = 0
+
+            #  request for namespaces is limited to 200 at a time, thus the request is fulfilled in increments of 200s.
+            while collecting_namespaces:
+                namespace_metadata = self.get_cloud_function_namespaces_metadata(region, offset)
+                if namespace_metadata['total_count'] == max_limit:
+                    offset += max_limit
+                else:
+                    collecting_namespaces = False
+
+                for name_space in namespace_metadata['namespaces']:
+                    if 'name' in name_space:  # API based namespace
+                        namespaces[name_space['name']] = {'type': 'API_based', 'id': name_space['id'],
+                                                          'region': name_space['location']}
+                    else:  # cloud foundry based namespace
+                        namespaces[name_space['id']] = {'type': 'CF_based', 'region': name_space['location']}
+
+        return namespaces
+
+    def create_cloud_function_namespaces(self, region):
+        """creates a name space in a given region, under a specified resource group and returns the namespace id"""
+
+        @spinner
+        def request_new_namespace():
+            print("Creating a new namespace...")
+            return requests.post(f'https://{region}.functions.cloud.ibm.com/api/v1/namespaces',
+                                 headers=headers, json=data).json()
+
+        iam_token = self.get_oauth_token()
+        resource_group_id = self.select_resource_group()
+        headers = {'Authorization': iam_token, 'accept': 'application/json'}
+
+        namespace_created = False
+        while not namespace_created:
+            try:
+                chosen_namespace = free_dialog("Please name your IBM cloud function namespace")['answer']
+                data = {"name": chosen_namespace, "resource_group_id": resource_group_id,
+                        "resource_plan_id": "functions-base-plan"}
+                response = request_new_namespace()
+
+                if 'id' not in response.keys():
+                    print("Chosen name for the namespace isn't valid. Server message:\n", response['message'])
```

Comment:
This one can be extracted to validate method and provided to inquirer as input validate attribute

Reply:
discussed offline. isn't related to inquirer.